magnolia-voting-app/src/main/java/com/example/votingapp/valueprovider/PollStatusValueProvider.java

```diff
@@ -0,0 +1,50 @@
+package com.example.votingapp.valueprovider;
+
+import java.util.*;
+
+import javax.jcr.*;
+
+import com.vaadin.data.*;
+
+import org.slf4j.*;
+
+/**
+ * PollStatusColumnDefinition
+ */
+public class PollStatusValueProvider implements ValueProvider<Node, String> {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(PollStatusValueProvider.class);
+
+    public static final String TO_DO = "To-do";
+    public static final String ON_GOING = "On-going";
+    public static final String DONE = "Done";
+
+    @Override
+    public String apply(Node source) {
+        try {
+            Calendar startDate = source.getProperty("startDate").getValue().getDate();
+            Calendar expiryDate = source.getProperty("expiryDate").getValue().getDate();
+            
+            return getPollStatus(startDate, expiryDate);
+
+        } catch (RepositoryException e) {
+            LOGGER.error("error", e);
```

Comment:
Add more details for error message.

Reply:
Return null is a bad practice, why don't we just wrap this exception to run time exception and let the container handle things? it's fast failing

Reply:
llamkhan: Yes. I have modified the code so as to follow your advice.